geth/peers/peerpool.go

```diff
@@ -95,22 +96,55 @@ func (p *PeerPool) Start(server *p2p.Server) error {
 // handleServerPeers watches server peer events, notifies topic pools about changes
 // in the peer set and stops the discv5 if all topic pools collected enough peers.
 func (p *PeerPool) handleServerPeers(server *p2p.Server, events <-chan *p2p.PeerEvent) {
+	var (
+		toSearch    []*TopicPool
+		retryDiscv5 <-chan time.Time
+	)
+	runListener := func() {
+		if server.DiscV5 == nil {
```

Comment:
When can it happen that `server.DiscV5` is `nil`?

Reply:
i set Discv5 to nil when it is stopped